_posts/2018-07-25-dota-macos-performance.md

```diff
@@ -0,0 +1,64 @@
+| OpenGL          | ?                | 105-110 fps  | 45.6 fps      | 5.7             | 33.9 fps | 6.8  |
+
+The first metric shows how much time of the main thread is spent inside the portability library, compared to the total execution time (which is the actual time minus all the sleeping). It was measured with Instruments on a simple scene ("fps - simple" contains the frame rate on it). Note that the submission is done on a separate thread by Dota, so that time isn't taken into account here. Interestingly, the less time we spend on the main thread the faster our frame rate ends up being. Or, in other words, it's the race of who gets to the submission first :)
+
+The "fps - complex" and "fps variability" are from the `Source2Bench.csv` produced by timing the [dota2-pts-1971360796.dem](http://www.phoronix-test-suite.com/benchmark-files/dota2-pts-1971360796.dem.tar.bz2) demo in range 40000 .. 50000. The "/low" group refers to `-autoconfig_level 1 -low` command line, while the "/high" group is for `-autoconfig_level 3 -high`. We followed the instructions provided by [GamingOnLinux](https://www.gamingonlinux.com/articles/want-to-benchmark-dota-2-on-linux-heres-how-to-do-it.7435). To be fair, we were only testing on the simple scene until this point, so more discoveries are possible once we start profiling this complex demo.
+
+#### Experiment Setup
+
+Hardware:
+  - MacBook Pro 2016
+  - 3.3 GHz Intel Core i7 (dual-core)
+  - Intel Iris Graphics 550
+
+Software: macOS 10.14 beta
+Resolution: 1440 x 900
+
+The benchmark scripts can be seen at our [gfx-portability PR](https://github.com/gfx-rs/portability/pull/118). We were running the same scene as [Phoronix](https://www.phoronix.com/scan.php?page=news_item&px=Dota-2-Initial-Mac-Vulkan), just with a longer time frame and slightly adjusted settings.
+
+### Conclusions
+
+MoltenVK does a good job translating Vulkan to Metal. However, benchmarking shows that it has subpar framerate variability (which is not desirable for a competitive game), and can even be slower than OpenGL on low settings. This doesn't match either Phoronix or MoltenVK's numbers, although we could think of the differences in the test setup that could explain that:
+  - _hardware_: ours has 2 physical cores, while Phoronix had 4
+  - _OS_: we are running on 10.14 beta, while Phoronix used 10.13.4
+  - _settings_: we run with `-vulkan_disable_occlusion_queries` since it's not yet implemented in our library
+
+Either way, our benchmarks show that OpenGL is fairly good on MacOS, and it's a high bar to beat for any Vulkan portability implementation, at least in the context of Dota performance. The change of tides on high settings could be explained by the compute shaders use by Dota, which are not available on MacOS's older OpenGL version. In other words, it's not the raw performance to gain on MacOS but rather more features, allowing certain workloads to use the GPU more efficiently.
```

Comment:
'The change of tides on high' could be 'The tides turning on high' if you want to keep the metaphor, or possibly slightly more blandly 'The performance shift seen on high'